### PR TITLE
Docs: update reproducible builds guide

### PR DESCRIPTION
**Pull request by duboism**
As discussed in #10924, we propose to update the reproducible builds guide (`docs/user/guides/reproducible-builds.rst`). The goals are:

  - update instruction for config v2 file
  - clarify that versions in the guide are just example and not recommended
  - don't use the buggy `sphinx=5.3.0` in the example

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--10939.org.readthedocs.build/en/10939/

- Developer's documentation (`dev`): https://dev--10939.org.readthedocs.build/en/10939/

### CODE REVIEW

**Comment by duboism** [comment]
I finished the main proposed modifications.

However, I don't understand the "✅ Good:" badge in **Pin your transitive dependencies**. Any thoughts ?

**Review by humitos — CHANGES_REQUESTED** [review]
Thanks for the PR! I think it's good and ready to merge after some small changes 👍🏼

File: docs/user/guides/reproducible-builds.rst [hunk 1]
```diff
@@ -53,9 +59,9 @@ A configuration file with explicit dependencies looks like this:
    :caption: docs/requirements.txt
 
    # Defining the exact version will make sure things don't break
-   sphinx==5.3.0
-   sphinx_rtd_theme==1.1.1
-   readthedocs-sphinx-search==0.1.1
+   sphinx==6.2.1
+   sphinx-rtd-theme==1.3.0
+   readthedocs-sphinx-search==0.3.1
```
[the remark this answers is not in the document]
Comment (duboism) [reply docs/user/guides/reproducible-builds.rst]: OK, I reverted the commits to restore the old versions.

File: docs/user/guides/reproducible-builds.rst [hunk 2]
```diff
@@ -12,15 +12,21 @@ so that you can focus on content.
    :local:
    :depth: 3
 
-Use a ``.readthedocs.yaml`` configuration file
-----------------------------------------------
+.. note::
+   This page explains how to pin the versions that correctly build your project.
+   The versions used in this page are just for illustrative purpose.
+   They are not the latest nor the recommended versions.
+   It is up to you to find those versions.
```
[the remark this answers is not in the document]
Comment (duboism) [reply docs/user/guides/reproducible-builds.rst:19]: Thanks :)

File: docs/user/guides/reproducible-builds.rst [hunk 3]
```diff
@@ -12,15 +12,21 @@ so that you can focus on content.
...
+This file controls how you documentation is build so it plays a crucial role for reproducible builds.
 
-This allows you to validate changes using :doc:`pull requests </pull-requests>`,
-and ensures that all your versions can be rebuilt from a reproducible configuration.
+Pin the tools version
+---------------------
+
+The configuration file is used to set the versions of the main tools used to build your project.
+For instance, you can require a specific version of Python.
+See the ``build.tools`` key in the documentation.
```
[the remark this answers is not in the document]
Comment (duboism) [reply docs/user/guides/reproducible-builds.rst:29]: OK. Given the previous remark (link to https://docs.readthedocs.io/en/stable/config-file/), we can transform the previous paragraph (Before reading this page) to an admonition just to mention that fixing the versions of the main tools in this file is a first step toward reproducibility.

**Comment by duboism** [comment]
Sorry for the delay. I commented on the various points you raised.

**Comment by duboism** [comment]
Any news ?

**Comment by duboism** [comment]
Still no news ?

**Comment by ericholscher** [comment]
This PR has conflicts -- I think because we addressed the change with the YAML file being required already. So it's probably fine to update this to just include the warning, and I'm fine to merge it.

**Comment by htech9** [comment]
@duboism You may want to fix the conflicts and address the comment above.

**Comment by duboism** [comment]
@htech9, sorry I was busy on other duties. I will adress that quickly.

**Comment by duboism** [comment]
As requested, I updated the branch to just include the warning (and an obvious typo).

**Review by ericholscher — APPROVED** [review]
Thanks :)

**Comment by duboism** [comment]
Thanks for merging. The world is a little better now :)

**Pull request merged**